dnf.spec

```diff
@@ -135,10 +135,11 @@ Provides:       dnf-command(search)
 Provides:       dnf-command(updateinfo)
 Provides:       dnf-command(upgrade)
 Provides:       dnf-command(upgrade-to)
+Recommends:     dnf-command(system-upgrade)
```

Comment:
I don't believe that this is good idea. In case that someone will remove the plugin for some reason, it will reappear every time when dnf will be updated. For same reason we do not recommend dnf-plugin-core.

Reply:
That isn't how `Recommends:` works, is it? I thought one of the main features of that directive is that the package should not get pulled in if the user has explicitly removed it. Are you suggesting that `Recommends:` only differs from requires in cases of conflict or a missing package?

Reply:
Yes, weak deps are not installed when package is not present, or cannot be installed or the feature is disabled in dnf configuration. The configuration is applicable as a global variable and cannot be set for a particular package.

Reply:
What @sgallagher describes is how many agree recommends should work. See https://bugzilla.redhat.com/show_bug.cgi?id=1699672

Reply:
@j-mracek, OK, so let's assume that you don't want this added upstream. I also submitted it downstream in Fedora at https://src.fedoraproject.org/rpms/dnf/pull-request/19 where I think it really needs to be included (and I considered treating it as `Requires:`, honestly. Since there's really no situation wherein we don't want that in Fedora.)

However, I am aware that DNF syncs its RPM from upstream, so I copied it here. Do you want me to change the patch to conditionalize it on the Fedora case?

I'll show my proposed change inline.

Reply:
@sgallagher What about to update comps. `dnf` is part of core group. You can add also python3-dnf-plugins-core or python3-dnf-plugin-system-upgrade there. Adding weak dep to dnf to install python3-dnf-plugin-system-upgrade is really not good idea. Using comps is more proper way, but still wrong one, because there are multiple deployment where they will not like it. Probably an other group would be more proper.

Reply:
@j-mracek I think you're missing the point I'm trying to make here: There are *zero* instances where we would not want `dnf-command(system-upgrade)` installed on any Fedora system. It is the only way to ensure continuity of updates. I really don't understand what argument you are trying to make here for not including it. When would it *ever* be desirable not to have it?

Even if the user would generally use GNOME-Software for updates, having this feature available as a fallback is highly desirable.

@AdamWill @kparal Would you mind chiming in here as a user-representative?

Reply:
I dunno, it feels like one of those xkcd 'corner case' things - *someone* is going to have a reason not to have it. they reinstall every release, they don't upgrade. they upgrade using dnf directly because they're EXTREEEEEEME. yanno.

dnf does specifically tell you exactly what to do if you try to run it when it's not installed, so it's not that terribly obnoxious not to have it. but at the same time having it installed is not that onerous. I am finding it hard to summon up strong feelings either way.

Reply:
Isn't that exactly the point of recommends?

Reply:
DNF itself recommends what to do.
`
$ sudo dnf system-upgrade asdsa
No such command: system-upgrade. Please use /usr/bin/dnf --help
It could be a DNF plugin command, try: "dnf install 'dnf-command(system-upgrade)'"
`

Reply:
I don't think dnf should _require_ system-upgrade. It might _recommend_ system-upgrade, but I don't have any hard feelings about it. Many end users don't need it at all, because they upgrade using gnome-software. If people feel it should be installed by default in Fedora, we can easily put it into Fedora comps, and we don't need to add any hard/weak deps. That also has the benefit that it can be uninstalled if you don't want it and it doesn't come back automatically. Similarly to Adam, my opinion is :man_shrugging:

Reply:
OK, so given that DNF's `Recommends:` do not work the way that I expected, I suppose the best option is indeed to just drop this portion of the patch for now. I'll get an updated patch out shortly.